nequip/train/metrics.py

```diff
@@ -184,16 +212,22 @@ def flatten_metrics(self, metrics, type_names=None):
         skip_keys = []
         for k, value in metrics.items():
 
-            key, reduction = k
+            key, param_hash = k
+            reduction, params = self.params[key][param_hash]
+
             short_name = ABBREV.get(key, key)
 
-            item_name = f"{short_name}_{reduction}"
+            per_atom = params["PerAtom"]
+            suffix = "/N" if per_atom else ""
+            item_name = f"{short_name}{suffix}_{reduction}"
 
-            stat = self.running_stats[key][reduction]
-            per_species = self.per_species[key][reduction]
+            stat = self.running_stats[key][param_hash]
+            per_species = params["PerSpecies"]
 
             if per_species:
                 if stat.output_dim == tuple():
+                    if type_names is None:
+                        type_names = [i for i in range(len(value))]
```

Comment:
is this a backwards compatability hack?

Reply:
yeah. I kept running in to the issue where `type_name` is none for `nequip-train minimal_eng.yaml`... Still don't know where exactly the bug is. But if we don't allow `type_name` to be none, we shouldn't let it to be an optional argument in the first place.

Reply:
... it should never be none?